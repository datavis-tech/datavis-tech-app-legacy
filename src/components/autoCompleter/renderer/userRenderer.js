import { Label } from 'semantic-ui-react'

export default ({ title }) => <Label content={title} />
