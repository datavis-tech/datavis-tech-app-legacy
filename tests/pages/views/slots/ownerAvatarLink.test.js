import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

jest.mock('../../../../src/components/avatarLink', () => () => <span>avatar</span>)

import AvatarLink from '../../../../src/components/avatarLink'
import OwnerAvatarLink from '../../../../src/pages/views/slots/ownerAvatarLink'

describe('owner avatar link', () => {

  let sut
  let user

  beforeEach(() => {
    user = Symbol('user')
    sut = shallow(<OwnerAvatarLink user={user} />)
  })

  it('should contain avatar link', () => {
    expect(sut.find(AvatarLink).prop('user')).toBe(user)
  })

  it('should be properly styled and formatted', () => {
    const tree = renderer
      .create(<OwnerAvatarLink user={user} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})
