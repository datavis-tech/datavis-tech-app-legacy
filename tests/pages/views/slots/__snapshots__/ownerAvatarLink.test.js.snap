// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`owner avatar link should be properly styled and formatted 1`] = `
<div
  style={
    Object {
      "fontSize": "0.8em",
      "marginBottom": "0.3em",
    }
  }
>
  <span
    style={
      Object {
        "marginRight": "3px",
      }
    }
  >
    By
  </span>
  <span>
    avatar
  </span>
</div>
`;
